### ISSUE TITLE
Implement ParkourRun Class

### ISSUE THREAD

**Issue by tmb** [hackcraft/parkour#1]
`ParkourRun` is a class that should represent a singular run on a parkour map. It should be associated with the Player who initiated it, and include a timing mechanism, the map they are playing on, and probably the index of the last checkpoint hit.

It should be able to be serialized somehow for storage into Redis.

It will be used from CheckpointListener to update the player's last hit checkpoint, verify that they have hit the previous checkpoint, etc..

It'll also be used to put a timer somewhere on the user screen (probably above the inventory)

**Comment by nice6599** [comment]
should there be a new instance for each attempt and we create another class for attempts or best scores or do we want to store best scores in a player data class

**Comment by nice6599** [comment]
i think this class should also include some methods that teleport the player to the start. that could be triggered by the item menus.

**Comment by tmb** [comment]
@nice6599 yeah, I think a new instance for each attempt is fine. It should probably have the time it took to complete the run.

we can get best scores by just iterating through a player's runs on a certain map and picking out the lowest time

**Comment by nice6599** [comment]
does creating a new instance for each run cause extra work for the garbage collector?
